compiler/luci/pass/src/FuseMulWithFullyConnectedPass.test.cpp

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/FuseMulWithFullyConnectedPass.h"
+#include "helpers/CreateCircleConst.h"
+
+#include <luci/IR/CircleNodes.h>
+#include <luci/test/TestIOGraph.h>
+
+#include <gtest/gtest.h>
+
+#define DIM_ONE 8
+#define DIM_TWO 4
+#define MUL_VAL 2.0f
+
+namespace
+{
+
+using namespace luci::test;
+
+/**
+ *  Graph for this test
+ *
+ *  BEFORE
+ *
+ *         [FC]
+ *           |
+ *     [Mul w/ Relu]
+ *
+ *  AFTER
+ *
+ *      [FC w/ Relu] (weights and bias updated)
+ *
+ */
+class FCMulGraphlet
+{
+public:
+  void init(loco::Graph *g, luci::FusedActFunc fc_activation, bool is_mul_scalar, bool use_bias)
+  {
+    _fc = g->nodes()->create<luci::CircleFullyConnected>();
+
+    std::vector<float> weights_val(DIM_ONE * DIM_TWO);
+    for (uint32_t i = 0; i < DIM_ONE * DIM_TWO; i++)
+      weights_val.at(i) = i;
+
+    _fc_f = luci::create_const_node(g, loco::DataType::FLOAT32, {DIM_ONE, DIM_TWO}, weights_val);
+    _fc->weights(_fc_f);
+
+    if (use_bias)
+    {
+      std::vector<float> bias_val(DIM_ONE);
+      for (uint32_t i = 0; i < DIM_ONE; i++)
+        bias_val.at(i) = i;
+
+      _fc_b = luci::create_const_node(g, loco::DataType::FLOAT32, {DIM_ONE}, bias_val);
+      _fc->bias(_fc_b);
+    }
+
+    _fc->fusedActivationFunction(fc_activation);
+    _fc->dtype(loco::DataType::FLOAT32);
+    _fc->shape({1, DIM_ONE});
+    _fc->name("fc");
+
+    std::vector<float> mul_values;
+
+    if (is_mul_scalar)
+    {
+      mul_values.push_back(static_cast<float>(MUL_VAL));
+      _mul_c = luci::create_const_node(g, loco::DataType::FLOAT32, {}, mul_values);
+    }
+    else
+    {
+      for (uint32_t i = 0; i < DIM_ONE; i++)
+      {
+        mul_values.push_back(static_cast<float>(i));
+      }
+      _mul_c = luci::create_const_node(g, loco::DataType::FLOAT32, {1, 1, 1, DIM_ONE}, mul_values);
+    }
+
+    _mul = g->nodes()->create<luci::CircleMul>();
+    _mul->x(_fc);
+    _mul->y(_mul_c);
+    _mul->fusedActivationFunction(luci::FusedActFunc::RELU);
+    _mul->dtype(loco::DataType::FLOAT32);
+    if (is_mul_scalar)
+    {
+      _mul->shape({1});
```

Comment:
```suggestion
      _mul->shape({1, DIM_ONE});
```

Reply:
As far as I understand the request in https://github.com/Samsung/ONE/pull/13607#discussion_r1713164640 , I am not sure about this one. This testcase is covering a scalar case where all values are broadcasted along the channels. Making it `{1, X}` is basically the same as `{1, 1, 1, X}`. Am I missing something?

Reply:
Shape of mul's inputs are `x={1, DIM_ONE}` and `y = scalar`. After broadcasting, the size of the result would be `{1, DIM_ONE}`. Could you check this?

Reply:
@jinevening You are absolutely right, I've got mixed up with another line. Adjusted!